src/AMQPRPCClient.js

```diff
@@ -1,122 +1,151 @@
 const Command = require('./Command');
 const CommandResult = require('./CommandResult');
+const AMQPEndpoint = require('./AMQPEndpoint');
 
 /**
  * This class is responsible for sending commands to the RPC server.
  *
  * @class
  */
-class AMQPRPCClient {
+class AMQPRPCClient extends AMQPEndpoint {
   /**
    * Creates a new instance of an RPC client.
    *
    * @param {*} connection Instance of `amqplib` library
-   * @param {String} exchange Exchange
-   * @param {String} key Key
-   * @param {Object} options Additional options for the client
-   * @param {Number} [options.timeout=60000] Timeout for cases when server is not responding
+   *
+   * @param {Object} params
+   * @param {String} params.requestsQueue queue for sending commands, should correspond with AMQPRPCServer
+   * @param {String} [params.repliesQueue=''] queue for feedback from AMQPRPCServer,
+   *    default is '' which means auto-generated queue name
+   * @param {Number} [params.timeout=60000] Timeout for cases when server is not responding
    */
-  constructor(connection, exchange, key, options) {
-    this._connection = connection;
-    this._exchange = exchange;
-    this._key = key;
-    this._options = Object.assign({timeout: AMQPRPCClient.TIMEOUT}, options);
-    this._channel = null;
+  constructor(connection, params = {}) {
+    params.repliesQueue = params.repliesQueue || '';
+    params.timeout = params.timeout || AMQPRPCClient.TIMEOUT;
+
+    super(connection, params);
+
+    this._repliesQueue = params.repliesQueue;
+    this._cmdNumber = 0;
+    this._requests = new Map();
   }
 
   /**
    * Send a command into RPC queue.
    *
    * @param {String} command Command name
-   * @param {Array<*>} args Array of any arguments provided to the RPC server callback
+   * @param [Array<*>] args Array of any arguments provided to the RPC server callback
    * @returns {Promise<*>}
    * @example
-   * client.sendCommand('some-command-name', [
-   *  {foo: 'bar'},
-   *  [1, 2, 3]
-   * ]);
+   * client.sendCommand('some-command-name', [{foo: 'bar'}, [1, 2, 3]]);
    */
   async sendCommand(command, args) {
-    return await this._sendCommand(new Command(command, args));
+    const cmd = new Command(command, args);
+
+    const correlationId = String(this._cmdNumber++);
```

Comment:
If more then one rpc client will use rpc server (handshake), then they may use same correlation tag. But it's not a problem.